### PR TITLE
(226) Initial terraform setup

### PR DESCRIPTION
**Pull request by Stretch96**
## Changes

This adds the initial terraform files needed for us to deploy to Azure
It also adds tests (using GitHub actions) to ensure the terraform code is valid
tfsec (https://github.com/aquasecurity/tfsec) is used as a security scanner for the terraform code, within another GitHub action

## Checklist

- [x] Attach this pull request to the appropriate card in Trello.
- [ ] Update the `CHANGELOG.md` if needed.

### CODE REVIEW

**Review by mec — APPROVED** [review]
All makes sense to me - I left a couple of thoughts/quesitons/nitpicks, nothing that should block this moving us forwards! 🚢

File: .github/workflows/tfsec.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: tfsec
+on:
+  pull_request:
+jobs:
+  tfsec:
+    name: PR commenter
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone repo
+        uses: actions/checkout@master
```
[the remark this answers is not in the document]
Comment (Stretch96) [reply .github/workflows/tfsec.yml]: Ta for picking this up, I've changed them all to `actions/checkout@v3` 👍

[the remark this answers is not in the document]
Comment (jacksonj04) [reply .github/workflows/tfsec.yml]: Followup: Is it worth adding GH actions to the dependabot config?

File: .github/workflows/tfsec.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      - name: tfsec
```
[the remark this answers is not in the document]
Comment (Stretch96) [reply .github/workflows/tfsec.yml:11]: Me and @mec had a discussion about the naming of the workflows/jobs, and will create a new PR for this

File: terraform/README.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
+This documentation covers the deployment of the infrastructure to host the app.
```
[the remark this answers is not in the document]
Comment (Stretch96) [reply terraform/README.md:1]: A link to this readme has been added to the root readme now 👍

File: .github/workflows/tfsec.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Stretch96) [reply .github/workflows/tfsec.yml]: Ah yep 👍 
(https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot)
I'll add a card to create a PR for that

**Pull request merged**